torch/testing/_asserts.py

```diff
@@ -102,6 +97,63 @@ def wrapper(actual: Tensor, expected: Tensor, **kwargs: Any) -> Optional[Excepti
     return wrapper
 
 
+def _check_quantization(check_tensor_values: Callable[..., Optional[Exception]]) -> Callable[..., Optional[Exception]]:
+    """Decorates continuous tensor check functions to handle quantized tensors.
+
+    If the inputs are not quantized, this decorator is a no-op.
+
+    Args:
+        check_tensor_values (Callable[..., Optional[Exception]]): Tensor check function for continuous tensors.
+
+    Returns:
+        Optional[Exception]: Return value of :attr:`check_tensors`.
+    """
+
+    @functools.wraps(check_tensor_values)
+    def wrapper(actual: Tensor, expected: Tensor, **kwargs: Any) -> Optional[Exception]:
+        if not actual.is_quantized:
+            return check_tensor_values(actual, expected, **kwargs)
+
+        if actual.qscheme() == torch.per_tensor_affine:
+            exc = check_tensor_values(torch.as_tensor(actual.q_scale()), torch.as_tensor(expected.q_scale()), **kwargs)
+            if exc:
+                return _amend_error_message(exc, "{}\n\nThe failure occurred for the quantization scale.")
+
+            exc = check_tensor_values(
+                torch.as_tensor(actual.q_zero_point()), torch.as_tensor(expected.q_zero_point()), **kwargs
+            )
+            if exc:
+                return _amend_error_message(exc, "{}\n\nThe failure occurred for the quantization zero point.")
+        elif actual.qscheme() == torch.per_channel_affine:
```

Comment:
could you add `torch.per_channel_affine_float_qparams` here? it has the same quantization parameters as `torch.per_channel_affine`, the only difference is the scales are float https://github.com/pytorch/pytorch/blob/master/torch/_tensor.py#L73

Reply:
Is this documented somewhere? I can't find anything in `master` docs.

Reply:
So I could simply add it to the same branch as `per_tensor_affine`? In the light of https://github.com/pytorch/pytorch/pull/58926#discussion_r652695746 should any parameters for this quantization scheme checked for closeness rather than equality?

Reply:
it should probably be in `torch.per_channel_affine` branch.
let me respond in the other comment about the closeness check